packages/react-components/react-teaching-popover-preview/src/components/TeachingPopover/useTeachingPopover.ts

```diff
@@ -0,0 +1,55 @@
+import { usePopover_unstable } from '@fluentui/react-popover';
+import type { TeachingPopoverProps, TeachingPopoverState } from './TeachingPopover.types';
+import * as React from 'react';
+import { TeachingPopoverCarousel } from '../TeachingPopoverCarousel/TeachingPopoverCarousel';
+import { useCallback } from 'react';
+
+export function countCarouselChildren(children: React.ReactElement | React.ReactElement[]): number {
+  const childArray = Array.isArray(children) ? children : [children];
+  let childCount = 0;
+  childArray.forEach((child: React.ReactElement) => {
+    if (!child || !child.props) {
+      return;
+    }
+    if (child.props && child.type === TeachingPopoverCarousel) {
+      childCount = React.Children.toArray(child.props.children).length;
+    } else if (child.props && child.props.children) {
+      const newChildCount = countCarouselChildren(child.props.children);
+      if (newChildCount > 0) {
+        childCount = newChildCount;
+      }
+    }
+  });
+
+  return childCount;
+}
+
+export const useTeachingPopover_unstable = (props: TeachingPopoverProps): TeachingPopoverState => {
+  const popoverState = usePopover_unstable(props);
+  const { onFinish, onPageChange } = props;
+
+  const carouselPageCount = useCallback(() => {
```

Comment:
@spmonahan just for my own knowledge, is there a specific reason for this or is it for stylistic/consistency reasons?

Reply:
It's largely stylistic/consistency but it also helps distinguish React hooks from custom hooks. I'll look into a lint.

Reply:
I briefly looked into a lint rule but didn't find anything (possibly I'm just searching with the wrong keywords). Since I don't have an imminent fix to enforce this I made an issue to follow up.

#30007